src/main/java/uk/gov/hmcts/reform/blobrouter/data/notifications/NotificationEnvelopeRepository.java

```diff
@@ -0,0 +1,31 @@
+package uk.gov.hmcts.reform.blobrouter.data.notifications;
+
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import java.util.List;
+
+@Repository
+public class NotificationEnvelopeRepository {
+
+    private final NamedParameterJdbcTemplate jdbcTemplate;
+    private final NotificationEnvelopeMapper mapper;
+
+    public NotificationEnvelopeRepository(
+        NamedParameterJdbcTemplate jdbcTemplate,
+        NotificationEnvelopeMapper mapper
+    ) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.mapper = mapper;
+    }
+
+    public List<NotificationEnvelope> getRejectedEnvelopes() {
+        return jdbcTemplate.query(
+            "SELECT env.file_name, env.container, event.type as error_type, event.notes as description "
+                + " FROM envelopes env, envelope_events event "
+                + " WHERE env.id = event.envelope_id "
+                + " AND env.status = 'REJECTED' AND event.type = 'REJECTED'",
```

Comment:
do we need both conditions ?

Reply:
envelope status can be removed but I thought it is clear this way.

Reply:
Also do we have index on status. I think we can only check on status/type and have an index ? what are your thoughts ?

Reply:
There is no index on status/type. I'll remove the status condition and add an index on `type`. is that ok?